src/packages/name-service/blockstack-service.ts

```diff
@@ -186,12 +186,13 @@ export class BlockstackService extends nameService.NameService {
     public registerName = async (identityClaim: nameService.IIdentityClaim, subdomain: string): Promise<string> => {
         const mnemonic = identityClaim.secrets.mnemonic;
         let identityKeyPair = identityClaim.secrets.identityKeyPair;
-        // Check for existing mnemonic
-        if (!mnemonic) {
-            throw ErrorHelper.getPackageError(PackageErrorCode.CouldNotFindMnemonicToRegisterName);
-        }
+
```

Comment:
Will it make sense to remove mnemonic from identityClaim?
If SDK is initialized with keyPair then identityClaim = keyPair
else: generate mnemonic and create a keyPair.. identityClaim = keyPair 

No reference to mnemonic in the code post init.